sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/NonAzureOpenAIKeyCredential.java

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.openai;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import reactor.core.publisher.Mono;
+
+import java.time.OffsetDateTime;
+
+/**
+ * Token credential class for accepting non-Azure OpenAI credential string.
+ */
+public class NonAzureOpenAIKeyCredential implements TokenCredential {
```

Comment:
This should just be called `OpenAIKeyCredential`.

Reply:
It used to call OpenAIKeyCredential but It is a suggested name taken from the service side after review. The reason is it will create a lot of confusion otherwise.  We can discuss it if needed.

Reply:
The scenario in our minds is that in the future we would have discussions such as:
"Hi, please don't use a NonAzureOpenAIKeyCredential with an Azure OpenAI URL" v/s "please don't use an OpenAIKeyCredential with Azure OpenAI". The latter being probably very confusing.